src/Illuminate/Foundation/Console/VendorPublishCommand.php

```diff
@@ -77,6 +78,27 @@ protected function publishTag($tag)
     }
 
     /**
+     * Determine the provider to publish.
+     *
+     * @return mixed
+     */
+    protected function providerToPublish()
+    {
+        if ($this->option('all')) {
+            return null;
+        }
+
+        if ($this->option('provider')) {
+            return $this->option('provider');
+        }
+
+        return $this->choice(
+            "Which package's files would you like to publish?",
+            ServiceProvider::providersAvailableToPublish()
+        );
```

Comment:
This might work
```
        $choice = $this->choice(
            "Which package's files would you like to publish?",
            array_merge(
                ['All'],
                ServiceProvider::providersAvailableToPublish()
            ),
            null
        );
        
        return $choice === 'All' ? null : $choice;
```